asteroid/masknn/convolutional.py

```diff
@@ -62,7 +62,7 @@ def forward(self, x):
         shared_out = self.shared_block(x)
         res_out = self.res_conv(shared_out)
         if not self.skip_out_chan:
-            return res_out
+            return res_out, res_out
```

Comment:
We should probably have a comment here to explain that this is for scriptability. 
I wonder, @faroit, is there a prettier way to do that? 
For example, this is not going to change when a model is exported (the control flow op is static), so this could be traced instead of scripted? And we could keep the same behaviour as currently?

Reply:
If you want to go branchless in the different `forward` methods, we could add a pointer to a function that switches to  the actual forward method when the class is initialized. Not necessarily the most elegant solution, but it should remove the need for `if` statements.

```python
    # In Conv1DBlock
    def forward(self, x):
        """ Input shape [batch, feats, seq]"""
        shared_out = self.shared_block(x)
        res_out = self.res_conv(shared_out)
        if not self.skip_out_chan:
            return res_out, res_out
        skip_out = self.skip_conv(shared_out)
        return res_out, skip_out
```

```python
    def __init__(self, ...):
        #...
        if self.skip_conv:
            self._forward = self._skip_out_forward
        else:
            self._forward = self._no_skip_forward

    def forward(self, x):
        """ Input shape [batch, feats, seq]"""
        return self._forward(x)

    def _common_forward(self, x):
        shared_out = self.shared_block(x)
        return self.res_conv(shared_out)

    def _skip_out_forward(self, x):
        res_out = self._common_forward(x)
        skip_out = self.skip_conv(shared_out)
        return res_out, skip_out

    def _no_skip_forward(self, x):
        res_out = self._common_forward(x)
        return res_out, torch.tensor([0.])
```

What do you think?

Reply:
But you needed to keep the `torch.tensor([0.])` in the `_no_skip_forward` because script would raise an error otherwise right?  I thought about that and I don't think it solves the problem. Scripting will raise the same typing issues and tracing will still work in the same way. 

Also, if we return two tensors, there is a simpler solution without if in the forward, which is much more readable IMO
```python
self.skip_conv = nn.Conv1d if skip_chan else nn.Identity()
...
return res_out, self.skip_conv(res_out)
```
but this still doesn't solve the problem. 

We can really on the dev to decide whether the control flow is static or not, if it's static, trace is fine, otherwise script is needed, and in this case, typing issues need to be fixed.

Reply:
Still very unsure about that..